Python/symtable.c

```diff
@@ -1581,7 +1581,12 @@ symtable_visit_stmt(struct symtable *st, stmt_ty s)
     }
     case TypeAlias_kind: {
         VISIT(st, expr, s->v.TypeAlias.name);
-        assert(s->v.TypeAlias.name->kind == Name_kind);
+        // This is only possible for a manually created AST
+        if (s->v.TypeAlias.name->kind != Name_kind) {
+            PyErr_SetString(PyExc_SyntaxError,
```

Comment:
This isn't the most detailed error message, but using `s->v.TypeAlias.name` with %R didn't work, I think it's not actually a Python object at this point. Given that you can only get here by manually creating an AST, I think the current error is fine.